sentry-tower/src/http.rs

```diff
@@ -0,0 +1,146 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use http_::Request;
+use tower_layer::Layer;
+use tower_service::Service;
+
+/// Tower Layer that logs Http Request Headers.
+///
+/// The Layer can also optionally start a new performance monitoring transaction
+/// based on incoming distributed tracing headers.
+#[derive(Clone, Default)]
+pub struct SentryHttpLayer {
+    start_transaction: bool,
+}
+
+impl SentryHttpLayer {
+    /// Creates a new Layer that only logs Request Headers.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Creates a new Layer that also starts a new performance monitoring transaction.
```

Comment:
this sounds like it always starts a new transaction rather than that it continues a transaction from any headers found (which is what the struct comment seems to suggest)